imake: Fix missing preInstall code

OPNA2608 opened the pull request:
Fixup to #402102, CC @quantenzitrone 

`x11BuildHook` only had a meaning when using `pkgs/servers/x11/xorg/builder.sh`, which would `source $x11BuildHook` if set. So this script goes unused now.

Not sourcing it causes `x11_ssh_askpass` to fail, because `$out/lib/X11/config` isn't getting populated anymore, and it expects to find some configs & templates there:
```
imake -DUseInstalled -I/nix/store/axnmg19pnx5jz68ap4c86wrr4403ghpk-imake-1.0.10/lib/X11/config
Imakefile.c:33:2: Include file Imake.tmpl not found
```
```
↪ env LANG=C ls -ahl /nix/store/axnmg19pnx5jz68ap4c86wrr4403ghpk-imake-1.0.10/lib
ls: cannot access '/nix/store/axnmg19pnx5jz68ap4c86wrr4403ghpk-imake-1.0.10/lib': No such file or directory
```

Drop `x11BuildHook`, and just add the `preInstall` code to the derivation directly.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [Nixpkgs 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/doc/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/doc/manual/release-notes/rl-2505.section.md) Nixpkgs Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
- [NixOS 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) NixOS Release notes)
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md), [pkgs/README.md](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md), [maintainers/README.md](https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md) and other contributing documentation in corresponding paths.

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

SuperSandro2000 commented on pkgs/by-name/im/imake/package.nix:
```diff
@@ -55,9 +55,15 @@ stdenv.mkDerivation (finalAttrs: {
 
+  preInstall = ''
+    mkdir -p $out/lib/X11/config
+    ln -s ${xorg-cf-files}/lib/X11/config/* $out/lib/X11/config
```
Shouldn't we cp them to avoid the extra dependency?

OPNA2608 replied:
`xorg-cf-files` is *just* all the text files that we're linking here. I don't think copying instead of linking to "avoid the extra dependency" makes any sense here if you're having size concerns, and it obscures where these configs actually come from when you look at the resulting `imake` files.

OPNA2608 commented on pkgs/by-name/im/imake/package.nix:
```diff
@@ -55,9 +55,15 @@ stdenv.mkDerivation (finalAttrs: {
+  preInstall = ''
+    mkdir -p $out/lib/X11/config
+    ln -s ${xorg-cf-files}/lib/X11/config/* $out/lib/X11/config
+    #touch $out/lib/X11/config/host.def # !!! hack
+    #touch $out/lib/X11/config/date.def # !!! hack
```
From https://github.com/NixOS/nixpkgs/commit/91e0b058af38c93a9267f8a82fa296e1af6fb336. It it hasn't come up again in the last 19 years, then it's prolly fine to drop this, sure.

OPNA2608 committed:
imake: Fix missing preInstall code

The builder.sh would previously source a specified x11BuildHook. Now that imake was pulled out from the big xorg code, there's nothing to actually handle a set x11BuildHook.

In imake's case, this hook added code to preInstall to symlink some config and template files from xorg-cf-files. This was causing x11_ssh_askpass to FTBFS.

Drop the separate file & x11BuildHook attribute, and put its code into the package file proper.

OPNA2608 commented:
@ofborg build x11_ssh_askpass

vcunat commented:
[xautolock](https://hydra.nixos.org/job/nixpkgs/staging-next/xautolock.x86_64-linux) is another example that gets fixed by this (and its reverse deps).

quantenzitrone reviewed (approved):
sorry for breaking stuff
